web/packages/teleport/src/Clusters/ManageCluster/ManageCluster.tsx

```diff
@@ -0,0 +1,212 @@
+/**
+ * Teleport
+ * Copyright (C) 2024  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React, { useCallback, useEffect, useState } from 'react';
+import styled from 'styled-components';
+import { Link } from 'react-router-dom';
+import { useParams } from 'react-router';
+
+import { useAsync, Attempt } from 'shared/hooks/useAsync';
+
+import { MultiRowBox, Row } from 'design/MultiRowBox';
+import Flex from 'design/Flex';
+import * as Icons from 'design/Icon';
+import Text, { H2 } from 'design/Text';
+import { Indicator } from 'design/Indicator';
+import Box, { BoxProps } from 'design/Box';
+
+import { LoadingSkeleton } from 'shared/components/UnifiedResources/shared/LoadingSkeleton';
+
+import { ShimmerBox } from 'design/ShimmerBox';
+import { Alert } from 'design/Alert';
+
+import {
+  FeatureBox,
+  FeatureHeader,
+  FeatureHeaderTitle,
+} from 'teleport/components/Layout';
+import { useTeleport } from 'teleport/index';
+import cfg from 'teleport/config';
+import { useNoMinWidth } from 'teleport/Main';
+import { ClusterInfo } from 'teleport/services/clusters';
+
+export function ManageCluster() {
+  const [cluster, setCluster] = useState<ClusterInfo>(null);
+  const ctx = useTeleport();
+
+  const { clusterId } = useParams<{
+    clusterId: string;
+  }>();
+
+  const [attempt, run] = useAsync(
+    useCallback(async () => {
+      const res = await ctx.clusterService.fetchClusterDetails(clusterId);
+      setCluster(res);
+      return res;
+    }, [clusterId, ctx.clusterService])
+  );
+
+  useEffect(() => {
+    if (!attempt.status && clusterId) {
+      run();
+    }
+  }, [attempt.status, run, clusterId]);
+
+  useNoMinWidth();
+
+  return (
+    <FeatureBox>
+      <ManageClusterHeader clusterId={clusterId} />
+      {attempt.status === 'processing' ? (
+        <Box textAlign="center" m={10}>
+          <Indicator />
+        </Box>
+      ) : (
+        <ClusterInformation cluster={cluster} attempt={attempt} />
+      )}
+    </FeatureBox>
+  );
+}
+
+export function ManageClusterHeader({ clusterId }: { clusterId: string }) {
```

Comment:
nit: un-export this? didn't see `ManageClusterHeader` usage beyond this file,

Reply:
This one is used on the teleport.e PR (see https://github.com/gravitational/teleport.e/pull/5231/files#diff-642b68829e407d1d215fa844f2c6895b5ebb3e8d975bb1da85fc98c04bf7ff03). Since we have to overwrite the component on Enterprise, I exported this and other components to maximize reuse between OSS and enterprise.